### ISSUE TITLE
Way to disable stdout capturing from outside of a test

### ISSUE THREAD

**Issue by dzcode** [pytest-dev/pytest#9775]
#### What's the problem this feature will solve?

If an env var is set, I'd like to open a port for remote debugging. In doing so, I'd like to prompt in the terminal that it is waiting to start, like this below, in a top-level `__init__.py` file of the app:
```py
import logging
import os
logger = logging.getLogger(__name__)
def setup_debugger():
    if os.environ.get("REMOTE_DEBUG"):
        import debugpy
        port = 5678
        logger.warning(f"Waiting for remote debugger to connect on port {port}")
        debugpy.listen(("0.0.0.0", port))
        debugpy.wait_for_client()
```
However, pytest swallows the output, so the terminal output is just hanging blank:
```
$ pytest

# hanging
```

#### Describe the solution you'd like

In a test, I'd typically be able to temporarily silence the output swallowing via the `capsys.disabled()`:
```
def test_disabling_capturing(capsys):
    print("this output is captured")
    with capsys.disabled():
        print("output not captured, going directly to sys.stdout")
    print("this output is also captured")
```

However, `capsys` appears only available as a test fixture, and I don't see a public API to disable stdout capture from outside of a test. I'd like a similar API to do so, e.g.:
```
from pytest import capsys

with capsys.disabled():
    logger.warning(f"Waiting for remote debugger to connect on port {port}")
```

See above in problem description

#### Alternative Solutions

I can run with the `-s` flag (or `--capture=tee-sys`), but that causes noisy output since I only care about printing errors from outside of test runs.

**Comment by taminomara** [comment]
I've ran into a similar issue. I'm trying to monkeypatch `sys.stderr` in a fixture, but pytest returns its value to the original before the test code. An option to disable this behavior on a per-test basis 
(or, in my case, from a fixture) would be nice!

**Comment by The-Compiler** [comment]
@taminomara That seems like a different situation to what OP describes in this issue. Have you tried doing

```python
@pytest.fixture
def patch_stderr(capsys, monkeypatch):
    # ... patching ...
    with capsys.disabled():
        yield
```

which, while I haven't tested it, should probably work?

**Comment by taminomara** [comment]
@The-Compiler yes, `capsys` re-enables capturing before each stage of a test run. I believe [this](https://github.com/pytest-dev/pytest/blob/8691ff1f368620c112983a5ef7c5ad82b46d9d34/src/_pytest/capture.py#L881) is the line where it happens.
